app/controllers/api/v1/project_preferences_controller.rb

```diff
@@ -3,35 +3,28 @@ class Api::V1::ProjectPreferencesController < Api::ApiController
 
   require_authentication :all, scopes: [:project]
   schema_type :json_schema
+  before_action :require_project_ownership, only: [:update_project_settings]
 
   def update_project_settings
-    binding.pry
-    project = Project.find params[:project_id]
-    unless required_params
-      render json: { error: 'missing or invalid params' }, status: :unprocessable_entity
-    end
-    if project.owner?(api_user.user)
-      upp = UserProjectPreference.find_or_initialize_by(project: project, user_id: api_user.user)
-      upp.settings.merge! params[:settings]
-      if upp.save!
-        render status: :ok, nothing: true
-      else
-        render status: :unprocessable_entity, nothing: true
-      end
+    upp = UserProjectPreference.find_or_initialize_by(project: project, user_id: api_user.user)
```

Comment:
hmm won't this only pull the requesting user's UPP instead of the desired upp of a user to update? Seem like you'll have to provide a `user_id` param to this route as well to load the correct upp.